pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -183,26 +183,7 @@ func validatePipelineContextVariables(tasks []PipelineTask) *apis.FieldError {
 	)
 	var paramValues []string
 	for _, task := range tasks {
-		var matrixParams []Param
-		var includeParams []Param
-		if task.IsMatrixed() {
-			matrixParams = task.Matrix.Params
-			if task.Matrix.hasInclude() {
-				for _, include := range task.Matrix.Include {
-					includeParams = include.Params
-				}
-			}
-		}
-		for _, param := range append(task.Params, matrixParams...) {
-			paramValues = append(paramValues, param.Value.StringVal)
-			paramValues = append(paramValues, param.Value.ArrayVal...)
-		}
-
-		if task.Matrix.hasInclude() {
-			for _, param := range append(task.Params, includeParams...) {
-				paramValues = append(paramValues, param.Value.StringVal)
-			}
-		}
+		paramValues = task.extractAllParamValues()
```

Comment:
```suggestion
		paramValues = task.extractAllParams().extractParamValues()
```

Reply:
note that this suggestion renames `params.extractParamValuesFromParams` to `params.extractParamValues` which I'd previously mentioned in https://github.com/tektoncd/pipeline/pull/6237#discussion_r1127089979

we can make this change in this refactor PR

Reply:
Renamed here